src/mempool_space/mod.rs

```diff
@@ -0,0 +1,47 @@
+pub mod api;
+pub mod websocket;
+
+use api::{BlockEvent, MempoolSpaceWebSocketRequestData, MempoolSpaceWebSocketRequestMessage};
+use bitcoin::Network;
+use futures_core::Stream;
+use url::Url;
+
+pub async fn subscribe_to_new_blocks(
+    network: &Network,
+    message: &MempoolSpaceWebSocketRequestMessage,
+) -> anyhow::Result<impl Stream<Item = BlockEvent>> {
+    let url: Url = url::Url::parse(&build_websocket_address(network)).unwrap();
+    websocket::connect_and_publish_message(url, message).await
+}
+
+pub fn build_websocket_request_message(
+    data: &MempoolSpaceWebSocketRequestData,
+) -> MempoolSpaceWebSocketRequestMessage {
+    let mut message = MempoolSpaceWebSocketRequestMessage {
+        action: String::from("want"),
+        data: vec![],
+    };
+
+    match data {
+        MempoolSpaceWebSocketRequestData::Blocks => message.data.push(String::from("blocks")),
+        MempoolSpaceWebSocketRequestData::MempoolBlocks => {
+            message.data.push(String::from("mempool-blocks"))
+        }
+        // FIXME: (@leonardo.lima) fix this track-address to use different struct
+        MempoolSpaceWebSocketRequestData::TrackAddress(..) => { /* ignore */ }
+    }
+    message
+}
+
+fn build_websocket_address(network: &Network) -> String {
```

Comment:
this could be renamed and made public so you could get the string and pass it in as I suggested above.

Reply:
@LLFourn I've made this one public and moved to websocket.rs, as it's specific for WebSocket, wdyt ?